source/particle/property/interface.cc

```diff
@@ -423,12 +423,17 @@ namespace aspect
                   // Determine if the current cell is at a Dirichlet boundary
                   bool cell_at_fixed_boundary = false;
                   unsigned int boundary_face = numbers::invalid_unsigned_int;
+                  double minimum_face_distance = std::numeric_limits<double>::max();
                   for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
                     if (cell->at_boundary(f) && fixed_boundaries.count(cell->face(f)->boundary_id()) == 1)
                       {
-                        boundary_face = f;
-                        cell_at_fixed_boundary = true;
-                        break;
+                        const double face_center_distance = particle_location.distance_square(cell->face(f)->center(true));
```

Comment:
This won't work well for cells that have an aspect ratio that's different from 1, but I don't really see a better solution.